contrib/ps.py

```diff
@@ -2,23 +2,282 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates.
 # SPDX-License-Identifier: LGPL-2.1-or-later
 
-"""A simplified implementation of ps(1) using drgn"""
+""" A simplified implementation of ps(1) using drgn """
 
-from drgn.helpers.common.format import number_in_binary_units
-from drgn.helpers.linux.mm import totalram_pages
-from drgn.helpers.linux.percpu import percpu_counter_sum
+import sys
+from argparse import ArgumentParser
+from collections import OrderedDict
+
+
+""" from drgn.helpers.common.format import number_in_binary_units """
+from drgn import ProgramFlags as pf
+from drgn.helpers.linux.cpumask import for_each_online_cpu
+from drgn.helpers.linux.list import list_for_each_entry
+from drgn.helpers.linux.mm import cmdline, totalram_pages
+from drgn.helpers.linux.percpu import per_cpu, percpu_counter_sum
 from drgn.helpers.linux.pid import for_each_task
 from drgn.helpers.linux.sched import task_cpu, task_state_to_char
 
 
+global cmd_opts
 PAGE_SIZE = prog["PAGE_SIZE"].value_()
 
 
+def get_number_of_children(task):
+    """ Returns number of children of a given task """
+
+    n = 0
+    for _ in list_for_each_entry("struct task_struct",
+            task.children.address_of_(), "sibling"):
+        n += 1
+    return n
+
+
+def get_cmd(task):
+    """ Return the commandline arguments of a given task """
+    cmd_args = ""
+    for cmd in cmdline(task):
+        cmd_args += cmd.decode() + " "
+
+    return cmd_args
+
+
+headers_and_vals = OrderedDict([
+    ("PID", lambda task: task.pid.value_()),
+    ("PPID", lambda task: task.parent.pid.value_()),
+    ("CPU", lambda task: task_cpu(task)),
+    ("Task Address", lambda task: hex(task.value_())),
+    ("Stack Address", lambda task: hex(task.stack.value_())),
+    ("State", lambda task: task_state_to_char(task)),
+    ("VMS", lambda task: PAGE_SIZE * task.mm.total_vm.value_()),
+    ("RSS", lambda task: get_rss(task)),
+    ("MEM%", lambda task: (round(((get_rss(task) * 100) /
+                           (PAGE_SIZE * totalram_pages(prog))), 4))),
+    ("comm", lambda task: task.comm.string_().decode())
+])
+
+headers_and_vals_detailed = OrderedDict([
+    ("Execution Time (sec)", lambda task: (task.utime + task.stime).value_()/1e9),
+    ("mm_struct addr", lambda task: hex(task.mm.value_())),
+    ("cmdline", get_cmd),
+    ("Number of children", get_number_of_children),
+    ("Stack Trace", lambda task: '\n' + str(prog.stack_trace(task.pid)))
+])
+
+
+def parse_cmdline_args(args):
+    """ Command line argument parser """
+
+    parser = ArgumentParser(prog="drgn ps",
+                            description="Report process status infromation")
+
+    parser.add_argument("-a", "--active", action="store_true", default=False,
+                        help="Print active thread on each CPU" +
+                        "(data may be inconsistent for live kernel)")
```

Comment:
This is missing a space, and you can let the string literals be concatenated without `+`:

```suggestion
                        help="Print active thread on each CPU "
                        "(data may be inconsistent for live kernel)")
```